src/libproc_macro/lib.rs

```diff
@@ -306,7 +307,7 @@ pub struct LineColumn {
 #[unstable(feature = "proc_macro", issue = "38356")]
 #[derive(Clone)]
 pub struct SourceFile {
-    filemap: Rc<FileMap>,
+    filemap: Lrc<FileMap>,
```

Comment:
Nah these are unstable and just private fields, so should be all good!

Reply:
Great, thanks for the quick feedback!

Reply:
This is IMO problematic because it will eventually be observable in user code - you should `impl !Send` and `!Sync` on `SourceFile` to hide the difference before we forget and stabilize it.

Reply:
There probably also other types this applies to in `libproc_macro`.